src/sutta_publisher/shared/github_handler.py

```diff
@@ -0,0 +1,43 @@
+import json
+import logging
+import os
+from base64 import b64encode
+from io import BytesIO
+
+import requests
+
+log = logging.getLogger(__name__)
+
+
+class GitHubHandler:
+    _REPO_URL = "https://api.github.com/repos/suttacentral/editions/contents/{file_name}"
+
+    @classmethod
+    def upload_file_to_repo(cls, file_path: str, file: BytesIO) -> None:
+        """Uploads given file to SuttaCentral's editions repository."""
+        file_bytes = file.read()
+        api_key = os.getenv("BOT_API_KEY")
+
+        if api_key is None:
+            raise ValueError("API key was not supplied!")
+
+        headers = {"Authorization": "token {api_key}"}
+        body = {
+            "message": "Uploading {file_path}",
+            "content": b64encode(file_bytes),
+            "sha": cls._get_file_sha(file_path),
+        }
+
+        request = requests.put(cls._REPO_URL.format(file_name=file_path), data=json.dumps(body), headers=headers)
+        request.raise_for_status()
+
+        log.info("** Publication uploaded to repo")
+
+    @classmethod
+    def _get_file_sha(cls, file_path: str) -> str:
+        """Return sha for existing file. If file does not exist return None."""
```

Comment:
if it's possible that we return None please fix return annotation: `-> str | None`